torch/nn/parallel/parallel_apply.py

```diff
@@ -61,7 +62,7 @@ def _worker(i, module, input, kwargs, device=None):
                 results[i] = output
         except Exception as e:
             with lock:
-                results[i] = e
+                results[i] = (e, sys.exc_info())
```

Comment:
This can cause reference cycle. See https://github.com/pytorch/pytorch/blob/c7448aa13c94ab47429484f4d02356c1a2f789ed/torch/utils/data/_utils/__init__.py#L18-L24

Reply:
Ah, good catch! Well, I can directly tack the stacktrace as a string, or create the new exception object I want to throw later. Or shall we reuse the `ExceptionWrapper` from your link here? Shall we move it to some common place?

Reply:
> to a common place

Any suggestion where that should be? I don't have a thorough overview of the code yet! Thanks!

Reply:
Let's put it in https://github.com/pytorch/pytorch/blob/master/torch/_utils.py!